src/sentry/api/endpoints/project_details.py

```diff
@@ -106,6 +106,14 @@ def _is_uniform_sampling_rule(rule):
         # future, we will have to update this to also support single transactions.
         return len(rule["condition"]["inner"]) == 0 and rule["type"] == "trace"
 
+    def validate_custom_rules_tag_names(self, rules):
+        # Ensure that all custom tags are valid name
+        # and are not presented in DS_DENYLIST
+        for rule in rules:
+            for op in rule["condition"]["inner"]:
+                if op["name"] in DS_DENYLIST:
```

Comment:
Is `name` always present here?

Reply:
@wedamija yeah i had similar thoughts, so for security let me add extra check.

@matejminar can you please help to figure out if `name` can be omitted.

Reply:
fixed in 640b796f16

Reply:
@andriisoldatenko frontend is always sending the name. what happens if someone fires the request without the name manually?